simulator/src/main/java/simblock/simulator/Simulator.java

```diff
@@ -46,6 +45,15 @@ public class Simulator {
 
   private static AbstractConsensusAlgo consensusAlgo = null;
 
+  private static int nextNodeIdx = 0;
+
+  public static Node getNextNode(){
+    Node nextNode = simulatedNodes.get(nextNodeIdx++ % simulatedNodes.size());
+    while(!nextNode.isReady()){
```

Comment:
can we please also check the condition if we loop through and there is none ready? In that case, we need to throw an exception for now.